Add missing icons + case-insensitive support

EntityPlantt opened the pull request:
### Resolves #3
## What's changed
* Added icon for `php`
* Added `html`, `jsm` and `njsproj` to the list

EntityPlantt committed:
Add missing icons connections

php, njsproj, jsm, html

EntityPlantt committed:
Add PHP logo

EntityPlantt committed:
Made 1 line of code simpler

a ? a : b
to
a || b

EntityPlantt committed:
Add case-insensitive support

minhazmiraz commented:
still not working. You have to add icon name in **materialFileIconsName.js**.
And change the php icon to this:

> php:
>     '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">\n<path d="M12 18.08c-6.63 0-12-2.72-12-6.08s5.37-6.08 12-6.08S24 8.64 24 12s-5.37 6.08-12 6.08m-5.19-7.95c.54 0 .91.1 1.09.31.18.2.22.56.13 1.03-.1.53-.29.87-.58 1.09-.28.22-.71.33-1.29.33h-.87l.53-2.76h.99m-3.5 5.55h1.44l.34-1.75h1.23c.54 0 .98-.06 1.33-.17.35-.12.67-.31.96-.58.24-.22.43-.46.58-.73.15-.26.26-.56.31-.88.16-.78.05-1.39-.33-1.82-.39-.44-.99-.65-1.82-.65H4.59l-1.28 6.58m7.25-8.33-1.28 6.58h1.42l.74-3.77h1.14c.36 0 .6.06.71.18.11.12.13.34.07.66l-.57 2.93h1.45l.59-3.07c.13-.62.03-1.07-.27-1.36-.3-.27-.85-.4-1.65-.4h-1.27L12 7.35h-1.44M18 10.13c.55 0 .91.1 1.09.31.18.2.22.56.13 1.03-.1.53-.29.87-.57 1.09-.29.22-.72.33-1.3.33h-.85l.5-2.76h1m-3.5 5.55h1.44l.34-1.75h1.22c.55 0 1-.06 1.35-.17.35-.12.65-.31.95-.58.24-.22.44-.46.58-.73.15-.26.26-.56.32-.88.15-.78.04-1.39-.34-1.82-.36-.44-.99-.65-1.82-.65h-2.75l-1.29 6.58z" fill="#1E88E5"/>\n</svg>'

minhazmiraz commented on src/components/icons/material/materialFileIcons6.js:
```diff
@@ -131,4 +131,6 @@ export const materialFileIcons6 = {
   credits:
     '<svg version="1.1" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">\n<path d="M3,3H21V5H3V3M7,7H17V9H7V7M3,11H21V13H3V11M7,15H17V17H7V15M3,19H21V21H3V19Z" fill="#9ccc65"/>\n</svg>',
+  php:
+    '<?xml version="1.0" encoding="UTF-8" standalone="no"?><svg height="383.5975" id="svg3430" version="1.1" viewBox="0 0 711.20123 383.5975" width="711.20123" xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:cc="http://creativecommons.org/ns#" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:svg="http://www.w3.org/2000/svg"><t ...
```
Change the icon to this:
`<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">\n<path d="M12 18.08c-6.63 0-12-2.72-12-6.08s5.37-6.08 12-6.08S24 8.64 24 12s-5.37 6.08-12 6.08m-5.19-7.95c.54 0 .91.1 1.09.31.18.2.22.56.13 1.03-.1.53-.29.87-.58 1.09-.28.22-.71.33-1.29.33h-.87l.53-2.76h.99m-3.5 5.55h1.44l.34-1.75h1.23c.54 0 .98-.06 1.33-.17.35-.12.67-.31.96-.58.24-.22.43-.46.58-.73.15-.26.26-.56.31-.88.16-.78.05-1.39-.33-1.82-.39-.44-.99-.65-1.82-.65H4.59l-1.28 6.58m7.25-8.33-1.28 6.58h1.42l.74-3.77h1.14c.36 0 .6.06.71.18.11.12.13.34.07.66l-.57 2.93h1.45l.59-3.07c.13-.62.03-1.07-.27-1.36-.3-.27-.85-.4-1.65-.4h-1.27L12 7.35h-1.44M18 10.13c.55 0 .91.1 1.09.31.18.2.22.56.13 1.03-.1.53-.29.87-.57 1.09-.29.22-.72.33-1.3.33h-.85l.5-2.76h1m-3.5 5.55h1.44l.34-1.75h1.22c.55 0 1-.06 1.35-.17.35-.12.65-.31.95-.58.24-.22.44-.46.58-.73.15-.26.26-.56.32-.88.15-.78.04-1.39-.34-1.82-.36-.44-.99-.65-1.82-.65h-2.75l-1.29 6.58z" fill="#1E88E5"/>\n</svg>`


Also add the file name in **materialFileIconsName.js**.

EntityPlantt replied:
Updated: https://github.com/EntityPlantt/file-extension-icon-JS/commit/02360cb08fa2d7020dbe5656c5cdfba8efeb42e8

minhazmiraz commented on src/components/api/MaterialIconApi.js:
```diff
@@ -7,6 +7,7 @@ import { materialFolderNamesToIcons } from "../gen/materialFolderIconsName";
 import Icons from "../icons/index";
 
 export const getMaterialFileIcon = (fileName) => {
+	fileName = fileName.toLowerCase();
```
Add this for VS icons too.

EntityPlantt replied:
Added in https://github.com/minhazmiraz/file-extension-icon-JS/pull/4/commits/1c8b603de41b956be4b3b306b8e9e7363c58eb8a

minhazmiraz commented:
Please try to test it before submitting the PR. Thank you 😊

EntityPlantt committed:
Update PHP logo

https://github.com/minhazmiraz/file-extension-icon-JS/pull/4/files/cc7517e3923218ba8d375cab3e41c3296e511e4b#r971440980

EntityPlantt committed:
Added case-insensitive support to VSI icons

EntityPlantt commented:
> **@minhazmiraz wrote:**
> Please try to test it before submitting the PR. Thank you 😊

Sorry, but I don't know how to build the package. (I've never built any NPM packages.)  
Can you reply with a step-to-step building process?

minhazmiraz commented:
> > **@minhazmiraz wrote:**
> > Please try to test it before submitting the PR. Thank you 😊
> 
> Sorry, but I don't know how to build the package. (I've never built any NPM packages.) Can you reply with a step-to-step building process?

1. Pull the code from github
2. run **npm install** in console
3. run **npm run start** to run the code in localhost

EntityPlantt commented:
> **@minhazmiraz wrote:**
> > > **@minhazmiraz wrote:**
> > > Please try to test it before submitting the PR. Thank you 😊
> > 
> > 
> > Sorry, but I don't know how to build the package. (I've never built any NPM packages.) Can you reply with a step-to-step building process?
> 
> 1. Pull the code from github
> 2. run **npm install** in console
> 3. run **npm run** to run the code in localhost

I get this error when running `npm start`:
```plain
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './lib/tokenize' is not defined by "exports" in C:\Users\NikTea\Documents\GitHub\file-extension-icon-JS\node_modules\postcss-safe-parser\node_modules\postcss\package.json
    at new NodeError (node:internal/errors:371:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:429:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:683:3)
    at resolveExports (node:internal/modules/cjs/loader:482:36)
    at Function.Module._findPath (node:internal/modules/cjs/loader:522:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:999:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Users\NikTea\Documents\GitHub\file-extension-icon-JS\node_modules\postcss-safe-parser\lib\safe-parser.js:1:17) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```
I've searched on the Internet, but it shows [0 results](https://www.google.com/search?q=Error+%5BERR_PACKAGE_PATH_NOT_EXPORTED%5D%3A+Package+subpath+%27.%2Flib%2Ftokenize%27+is+not+defined+by+%22exports%22+in+C%3A%5CUsers%5CNikTea%5CDocuments%5CGitHub%5Cfile-extension-icon-JS%5Cnode_modules%5Cpostcss-safe-parser%5Cnode_modules%5Cpostcss%5Cpackage.json&rlz=1C1LOQA_enMK915MK915&sourceid=chrome&ie=UTF-8).  
What can I do about it?

Tried running `npm test`, it shows a testing dialog.  
Also tried `npm run`, `npm run build` and `num run eject`. None of them works.

EntityPlantt commented:
And also one thing, while I'm at it: Change this **folder** word to **file**

![image](https://user-images.githubusercontent.com/92634082/190706138-68d300a7-dfc9-4340-aa8d-ba50ccb13527.png)

EntityPlantt committed:
Resolved https://github.com/minhazmiraz/file-extension-icon-JS/pull/4#issuecomment-1247471395

minhazmiraz commented:
> > > **@minhazmiraz wrote:**
> > > Please try to test it before submitting the PR. Thank you 😊
> > 
> > 
> > Sorry, but I don't know how to build the package. (I've never built any NPM packages.) Can you reply with a step-to-step building process?
> 
> 1. Pull the code from github
> 2. run **npm install** in console
> 3. run **npm run start** to run the code in localhost

Sorry, it should be **npm run start**.
